src/core/components/NewInstallButton/index.js

```diff
@@ -0,0 +1,336 @@
+  INSTALL_ACTION,
+  INSTALL_STARTED_ACTION,
+  UNINSTALLING,
+  UNKNOWN,
+} from 'core/constants';
+import translate from 'core/i18n/translate';
+import { findInstallURL } from 'core/installAddon';
+import log from 'core/logger';
+import { getThemeData } from 'core/themeInstall';
+import tracking, {
+  getAddonTypeForTracking,
+  getAddonEventCategory,
+} from 'core/tracking';
+import { isTheme } from 'core/utils';
+import AnimatedIcon from 'ui/components/AnimatedIcon';
+import Button from 'ui/components/Button';
+import Icon from 'ui/components/Icon';
+import type { AppState } from 'amo/store';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { ReactRouterLocationType } from 'core/types/router';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+  className?: string,
+  defaultInstallSource: string,
+  disabled: boolean,
+  location: ReactRouterLocationType,
+  puffy: boolean,
+  status: string,
+  // From `withInstallHelpers()`, see: `src/core/installAddon.js`.
+  enable: () => Promise<any>,
+  hasAddonManager: boolean,
+  install: () => Promise<any>,
+  installTheme: (HTMLAnchorElement, Object) => Promise<any>,
+  uninstall: (Object) => Promise<any>,
+|};
+
+type InternalProps = {|
+  ...Props,
+  _config: typeof config,
+  _log: typeof log,
+  _tracking: typeof tracking,
+  _window: typeof window,
+  i18n: I18nType,
+  userAgentInfo: string,
+|};
+
+type TrackParams = {|
+  addonName: string,
+  type: string,
+|};
+
+type ButtonProps = {|
+  buttonType: string,
+  className: string,
+  'data-browsertheme'?: string,
+  disabled: boolean,
+  href: string,
+  onClick: Function | null,
+  puffy: boolean,
+|};
+
+export class NewInstallButtonBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    _config: config,
+    _log: log,
+    _tracking: tracking,
+    _window: typeof window !== 'undefined' ? window : {},
+    puffy: true,
+  };
+
+  installTheme = (event: SyntheticEvent<HTMLAnchorElement>) => {
+    const { addon, status, installTheme } = this.props;
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    installTheme(event.currentTarget, { ...addon, status });
+  };
+
+  installOpenSearch = (event: SyntheticEvent<HTMLAnchorElement>) => {
+    const { _log, _window, addon } = this.props;
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    const installURL = event.currentTarget.href;
+
+    _log.info('Adding OpenSearch Provider', { addon });
+    _window.external.AddSearchProvider(installURL);
+
+    this.trackInstallStarted({
+      addonName: addon.name,
+      type: addon.type,
+    });
+
+    return false;
+  };
+
+  installExtension = async (event: SyntheticEvent<HTMLAnchorElement>) => {
+    const { addon, enable, install } = this.props;
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    await install();
+
+    if (addon.type === ADDON_TYPE_STATIC_THEME) {
+      await enable();
```

Comment:
I think we want to do something similar with the old set up too (in case this isn't turned on right away). Right now it only installs without enabling..

Reply:
I suppose this should be done in a separate PR then.

Reply:
sure, we have the header click too anyway. It could be a part of that work..